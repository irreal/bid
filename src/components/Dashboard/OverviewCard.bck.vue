<template>
  <v-card height="80" class="mt-4">
    <div class="title">
      <v-card
        ><v-icon>{{ icon }}</v-icon></v-card
      >
    </div>
    <v-card-text>Ovo je kartica</v-card-text>
  </v-card>
  <!-- <div class="overviewCard" @click="$emit('click')">
    <div class="overviewCard-icon overviewCard-icon--document">
      <i class="fas" :class="{ [icon]: icon }"></i>
    </div>
    <div class="overviewCard-description">
      <h3 class="overviewCard-title text-light">
        {{ text }}
        <strong>{{ textHighlight }}</strong>
      </h3>
      <p class="overviewCard-subtitle">{{ subtitle }}</p>
    </div>
  </div> -->
</template>

<style scoped lang="scss">
.title {
  width: 70px;
  height: 70px;
  position: relative;
  left: 25px;
  top: -10px;
}
</style>

<script>
export default {
  props: {
    icon: String,
    text: String,
    textHighlight: String,
    subtitle: String
  }
};
</script>
