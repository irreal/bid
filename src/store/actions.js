export default {
  updateUser: ({ commit }, payload) => {
    commit("setUser", payload.currentUser);
  }
};
