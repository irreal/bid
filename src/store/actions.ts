import { ActionTree } from 'vuex';
import State from './state';

const actions: ActionTree<State, State> = {
    //   updateColor({ commit }: ActionContext<State, State>, newColor: string): void {
    //     commit(COLOR, newColor);
    //   },
    };

export default actions;
