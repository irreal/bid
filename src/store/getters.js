export default {
  getUserStatus: state => !!state.user,
  getUser: state => state.user
};
