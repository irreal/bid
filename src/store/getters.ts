import { GetterTree } from 'vuex';
import State from './state';
import Plan from '@/models/plan';

const getters: GetterTree<State, State> = {
    plans(state: State): Plan[] {
      return state.plans;
    },
    // colour(state: State): string {
    // 	return state.color;
    // },
  };

export default getters;
