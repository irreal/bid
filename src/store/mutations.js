export default {
  setUser: (state, currentUser) => {
    state.user = currentUser;
  }
};
