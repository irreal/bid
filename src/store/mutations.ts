import { MutationTree } from 'vuex';
import State from './state';

const mutations: MutationTree<State> = {
    removePlans(state: State) {
      state.plans = [];
    },
    // color(state: State, newColor: string): void {
    // 	state.color = newColor;
    // }
  };

export default mutations;
