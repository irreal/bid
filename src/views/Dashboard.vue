<template>
  <div>
    <div class="main-header">
      <div class="main-header__intro-wrapper">
        <div class="main-header__welcome">
          <div class="main-header__welcome-title text-light">Zdravo,
            <strong>Žarko</strong>
          </div>
          <div class="main-header__welcome-subtitle text-light"></div>
        </div>
        <div class="quickview">
          <div class="quickview__item">
            <div class="quickview__item-total">10</div>
            <div class="quickview__item-description">
              <i class="far fa-file-alt"></i>
              <span class="text-light">Izveštaja</span>
            </div>
          </div>
          <div class="quickview__item">
            <div class="quickview__item-total">0</div>
            <div class="quickview__item-description">
              <i class="far fa-comment"></i>
              <span class="text-light">Poruka</span>
            </div>
          </div>
          <div class="quickview__item">
            <div class="quickview__item-total">12&deg;</div>
            <div class="quickview__item-description">
              <i class="fas fa-map-marker-alt"></i>
              <span class="text-light">Beograd</span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="main-overview">
      <div class="overviewCard">
        <div class="overviewCard-icon overviewCard-icon--document">
          <i class="fas fa-chart-line"></i>
        </div>
        <div class="overviewCard-description">
          <h3 class="overviewCard-title text-light">Novi
            <strong>Plan</strong>
          </h3>
          <p class="overviewCard-subtitle">plan prodaje u periodu</p>
        </div>
      </div>
      <div class="overviewCard">
        <div class="overviewCard-icon overviewCard-icon--calendar">
          <i class="far fa-calendar-check"></i>
        </div>
        <div class="overviewCard-description">
          <h3 class="overviewCard-title text-light">Sledeći
            <strong>Sastanci</strong>
          </h3>
          <p class="overviewCard-subtitle">kalendar timskih sastanaka</p>
        </div>
      </div>
      <!-- <div class="overviewCard">
        <div class="overviewCard-icon overviewCard-icon--mail">
          <i class="far fa-envelope"></i>
        </div>
        <div class="overviewCard-description">
          <h3 class="overviewCard-title text-light">Recent
            <strong>Emails</strong>
          </h3>
          <p class="overviewCard-subtitle">+10</p>
        </div>
      </div>
      <div class="overviewCard">
        <div class="overviewCard-icon overviewCard-icon--photo">
          <i class="far fa-file-image"></i>
        </div>
        <div class="overviewCard-description">
          <h3 class="overviewCard-title text-light">New
            <strong>Album</strong>
          </h3>
          <p class="overviewCard-subtitle">House Concert</p>
        </div>
      </div> -->
    </div>
    <!-- /.main__overview -->
    <div class="main__cards">
      <!-- <div class="card">
        <div class="card__header">
          <div class="card__header-title text-light">
            Your
            <strong>Events</strong>
            <a href="#" class="card__header-link text-bold">View All</a>
          </div>
          <div class="settings">
            <div class="settings__block">
              <i class="fas fa-edit"></i>
            </div>
            <div class="settings__block">
              <i class="fas fa-cog"></i>
            </div>
          </div>
        </div>
        <div class="card__main">
          <div class="card__row">
            <div class="card__icon">
              <i class="fas fa-gift"></i>
            </div>
            <div class="card__time">
              <div>today</div>
            </div>
            <div class="card__detail">
              <div class="card__source text-bold">Jonathan G</div>
              <div class="card__description">Going away party at 8:30pm. Bring a friend!</div>
              <div class="card__note">1404 Gibson St</div>
            </div>
          </div>
          <div class="card__row">
            <div class="card__icon">
              <i class="fas fa-plane"></i>
            </div>
            <div class="card__time">
              <div>Tuesday</div>
            </div>
            <div class="card__detail">
              <div class="card__source text-bold">Matthew H</div>
              <div class="card__description">Flying to Bora Bora at 4:30pm</div>
              <div class="card__note">Delta, Gate 27B</div>
            </div>
          </div>
          <div class="card__row">
            <div class="card__icon">
              <i class="fas fa-book"></i>
            </div>
            <div class="card__time">
              <div>Thursday</div>
            </div>
            <div class="card__detail">
              <div class="card__source text-bold">National Institute of Science</div>
              <div
                class="card__description"
              >Join the institute for an in-depth look at Stephen Hawking</div>
              <div class="card__note">7:30pm, Carnegie Center for Science</div>
            </div>
          </div>
          <div class="card__row">
            <div class="card__icon">
              <i class="fas fa-heart"></i>
            </div>
            <div class="card__time">
              <div>Friday</div>
            </div>
            <div class="card__detail">
              <div class="card__source text-bold">24th Annual Heart Ball</div>
              <div class="card__description">Join us and contribute to your favorite local charity.</div>
              <div class="card__note">6:45pm, Austin Convention Ctr</div>
            </div>
          </div>
          <div class="card__row">
            <div class="card__icon">
              <i class="fas fa-heart"></i>
            </div>
            <div class="card__time">
              <div>Saturday</div>
            </div>
            <div class="card__detail">
              <div class="card__source text-bold">Little Rock Air Show</div>
              <div class="card__description">See the Blue Angels fly with roaring thunder</div>
              <div class="card__note">11:00pm, Jacksonville Airforce Base</div>
            </div>
          </div>
        </div>
      </div> -->
      <div class="card">
        <div class="card__header">
          <div class="card__header-title text-light">
            Poslednji
            <strong>Izveštaji </strong>
            <a href="#" class="card__header-link text-bold">Pregledaj sve</a>
          </div>
          <div class="settings">
            <div class="settings__block">
              <i class="fas fa-edit"></i>
            </div>
            <div class="settings__block">
              <i class="fas fa-cog"></i>
            </div>
          </div>
        </div>
        <div class="card">
          <div class="documents">
            <div class="document">
              <div class="document__img"></div>
              <div class="document__title">realizacija</div>
              <div class="document__date">03.03.2019</div>
            </div>
            <div class="document">
              <div class="document__img"></div>
              <div class="document__title">budžet</div>
              <div class="document__date">24.02.2019</div>
            </div>
            <div class="document">
              <div class="document__img"></div>
              <div class="document__title">zalihe</div>
              <div class="document__date">10.02.2019</div>
            </div>
            <div class="document">
              <div class="document__img"></div>
              <div class="document__title">dospela potraživanja</div>
              <div class="document__date">03.02.2019</div>
            </div>
          </div>
        </div>
      </div>
      <div class="card card--finance">
        <div class="card__header">
          <div class="card__header-title text-light">
            Plan
            <strong>prodaje </strong>
            <a href="#" class="card__header-link text-bold">Pregledaj sve</a>
          </div>
          <div class="settings">
            <div class="settings__block">
              <i class="fas fa-edit"></i>
            </div>
            <div class="settings__block">
              <i class="fas fa-cog"></i>
            </div>
          </div>
        </div>
        <div id="chartdiv">
              <apexchart width="350" type="radialBar" :options="options" :series="series"></apexchart>
        </div>
      </div>
    </div>
    <!-- /.main-cards -->
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
// import BrandHeader from '@/components/BrandHeader.vue';
import { ApexOptions } from 'apexcharts';
@Component({
  //   components: {BrandHeader},
})
export default class Dashboard extends Vue {
  private options: ApexOptions = {
    plotOptions: {
      radialBar: {
        dataLabels: {
          total: {
            show: true,
            label: 'Ukupno:',
            formatter: () => {
              return `${Math.round(
                this.series.reduce((tot, cur) => tot + cur, 0) /
                  this.series.length,
              )}%`;
            },
          },
        },
      },
    },
    chart: {
      id: 'vuechart-example',
    },
    labels: ['Godina', 'Kvartal', 'Mesec'],
  };

  private series: ApexNonAxisChartSeries = [25, 70, 90];
}
</script>

<style scoped lang="scss">
</style>
