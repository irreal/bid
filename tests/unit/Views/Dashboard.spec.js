import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Dashboard from "@/views/Dashboard";
import Vuetify from "vuetify";
import Vue from "vue";
import initialState from "@/store/state";
import actions from "@/store/actions";
import getters from "@/store/getters";
import mutations from "@/store/mutations";
import userFixture from "../fixtures/user";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

describe("Dashboard Page", () => {
  let state;

  const build = () => {
    const dashboard = shallowMount(Dashboard, {
      localVue,
      store: new Vuex.Store({ state, actions, getters, mutations })
    });

    return {
      dashboard
    };
  };

  beforeEach(() => {
    state = { ...initialState };
  });
  it("renders the component", () => {
    // arrange
    const { dashboard } = build();
    // assert
    expect(dashboard.html()).toMatchSnapshot();
  });

  it("renders the component with a logged in user", () => {
    // arrange
    state.user = userFixture;
    const { dashboard } = build();
    // assert
    expect(dashboard.html()).toMatchSnapshot();
  });
});
