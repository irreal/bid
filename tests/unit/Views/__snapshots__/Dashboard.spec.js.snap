// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page renders the component 1`] = `
<div>
  <div>
    <v-parallax-stub height="180" alt="" src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/1609106/lake-shadow-water.jpg">
      <v-layout-stub tag="div" wrap="">
        <welcome-stub></welcome-stub>
        <quickview-stub class="hidden-sm-and-down"></quickview-stub>
      </v-layout-stub>
    </v-parallax-stub>
  </div>
  <div class="main-overview">
    <v-layout-stub tag="div" justify-space-around="" wrap="">
      <overviewcard-stub color="accent" elevation="10" fullwidth="true" offset="24">
        <h4 class="title font-weight-light">Pregled realizacije</h4>
        <p class="category d-inline-flex font-weight-light">
          Realizacija periodičnih planova prodaje
        </p>
      </overviewcard-stub>
      <overviewcard-stub color="accent" elevation="10" fullwidth="true" offset="24" icon="fa-chart-line">
        <h4 class="title font-weight-light">Pregled realizacije</h4>
        <p class="category d-inline-flex font-weight-light">
          Realizacija periodičnih planova prodaje
        </p>
      </overviewcard-stub>
      <overviewcard-stub color="accent" elevation="10" fullwidth="true" offset="24" icon="fa-chart-line">
        <h4 class="title font-weight-light">Pregled realizacije</h4>
        <p class="category d-inline-flex font-weight-light">
          Realizacija periodičnih planova prodaje
        </p>
      </overviewcard-stub>
    </v-layout-stub>
  </div>
  <div class="main__cards">
    <div class="card">
      <div class="card__header">
        <div class="card__header-title text-light">
          Poslednji
          <strong>Izveštaji&nbsp;</strong> <a href="#" class="card__header-link text-bold">Pregledaj sve</a></div>
        <div class="settings">
          <div class="settings__block"><i class="fas fa-edit"></i></div>
          <div class="settings__block"><i class="fas fa-cog"></i></div>
        </div>
      </div>
      <div class="card">
        <div class="documents">
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">realizacija</div>
            <div class="document__date">03.03.2019</div>
          </div>
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">budžet</div>
            <div class="document__date">24.02.2019</div>
          </div>
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">zalihe</div>
            <div class="document__date">10.02.2019</div>
          </div>
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">dospela potraživanja</div>
            <div class="document__date">03.02.2019</div>
          </div>
        </div>
      </div>
    </div>
    <div class="card card--finance">
      <div class="card__header">
        <div class="card__header-title text-light">
          Plan
          <strong>prodaje&nbsp;</strong> <a href="" class="card__header-link text-bold">Pregledaj sve</a></div>
        <div class="settings">
          <div class="settings__block"><i class="fas fa-edit"></i></div>
          <div class="settings__block"><i class="fas fa-cog"></i></div>
        </div>
      </div>
      <div id="chartdiv"></div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Page renders the component with a logged in user 1`] = `
<div>
  <div>
    <v-parallax-stub height="180" alt="" src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/1609106/lake-shadow-water.jpg">
      <v-layout-stub tag="div" wrap="">
        <welcome-stub></welcome-stub>
        <quickview-stub class="hidden-sm-and-down"></quickview-stub>
      </v-layout-stub>
    </v-parallax-stub>
  </div>
  <div class="main-overview">
    <v-layout-stub tag="div" justify-space-around="" wrap="">
      <overviewcard-stub color="accent" elevation="10" fullwidth="true" offset="24">
        <h4 class="title font-weight-light">Pregled realizacije</h4>
        <p class="category d-inline-flex font-weight-light">
          Realizacija periodičnih planova prodaje
        </p>
      </overviewcard-stub>
      <overviewcard-stub color="accent" elevation="10" fullwidth="true" offset="24" icon="fa-chart-line">
        <h4 class="title font-weight-light">Pregled realizacije</h4>
        <p class="category d-inline-flex font-weight-light">
          Realizacija periodičnih planova prodaje
        </p>
      </overviewcard-stub>
      <overviewcard-stub color="accent" elevation="10" fullwidth="true" offset="24" icon="fa-chart-line">
        <h4 class="title font-weight-light">Pregled realizacije</h4>
        <p class="category d-inline-flex font-weight-light">
          Realizacija periodičnih planova prodaje
        </p>
      </overviewcard-stub>
    </v-layout-stub>
  </div>
  <div class="main__cards">
    <div class="card">
      <div class="card__header">
        <div class="card__header-title text-light">
          Poslednji
          <strong>Izveštaji&nbsp;</strong> <a href="#" class="card__header-link text-bold">Pregledaj sve</a></div>
        <div class="settings">
          <div class="settings__block"><i class="fas fa-edit"></i></div>
          <div class="settings__block"><i class="fas fa-cog"></i></div>
        </div>
      </div>
      <div class="card">
        <div class="documents">
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">realizacija</div>
            <div class="document__date">03.03.2019</div>
          </div>
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">budžet</div>
            <div class="document__date">24.02.2019</div>
          </div>
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">zalihe</div>
            <div class="document__date">10.02.2019</div>
          </div>
          <div class="document">
            <div class="document__img"></div>
            <div class="document__title">dospela potraživanja</div>
            <div class="document__date">03.02.2019</div>
          </div>
        </div>
      </div>
    </div>
    <div class="card card--finance">
      <div class="card__header">
        <div class="card__header-title text-light">
          Plan
          <strong>prodaje&nbsp;</strong> <a href="" class="card__header-link text-bold">Pregledaj sve</a></div>
        <div class="settings">
          <div class="settings__block"><i class="fas fa-edit"></i></div>
          <div class="settings__block"><i class="fas fa-cog"></i></div>
        </div>
      </div>
      <div id="chartdiv"></div>
    </div>
  </div>
</div>
`;
